### PR TITLE
Add inline example for grdcut

### PR DESCRIPTION
**Pull request by willschlitzer**
This PR adds an inline code example to the docstring for `grdcut`.

Addresses #1686

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

File: pygmt/src/grdcut.py [hunk 1]
```diff
@@ -87,6 +87,18 @@ def grdcut(grid, **kwargs):
...
+    --------
+    >>> import pygmt
+    >>> # Load a grid of @earth_relief_30m data, with an x-range of 10 to 30,
+    >>> # and a y-range of 15 to 25
+    >>> grid = pygmt.datasets.load_earth_relief(
+    ...     resolution="30m", region=[10, 30, 15, 25]
+    ... )
+    >>> # Create a new grid from an input grid, with an x-range of -3 to 1,
+    >>> # and a y-range of 2 to 5
+    >>> new_grid = pygmt.grdcut(grid=grid, region=[-3, 1, 2, 5])
```
Comment (meghanrjones) [inline pygmt/src/grdcut.py:101]: Thanks for moving forward with the in-line examples!

I would prefer that the example using a different dataset than the result from `load_earth_relief`. We could add a raster dataset to the proposed `load_sample_data` function (https://github.com/GenericMappingTools/pygmt/pull/1685). I think that `grid = load_sample_data(name="relief")` (hypothetically for `@tut_relief`) is more intuitive than relying on`load_earth_relief`.

Comment (seisman) [reply pygmt/src/grdcut.py:101]: > I think that `grid = load_sample_data(name="relief")` (hypothetically for `@tut_relief.nc`) is more intuitive than relying on`load_earth_relief`.

Why do you think so? I think cutting a earth relief grid is a very common case, and users can simply use it without changes.

Comment (willschlitzer) [reply pygmt/src/grdcut.py:101]: I agree with @seisman on this. I think the examples are good to demonstrate how a user could get the code running, and it's better to demonstrate using the `load_earth_relief` for a usable example.

Comment (meghanrjones) [inline pygmt/src/grdcut.py:101]:
```suggestion
    >>> # Create a new grid from an input grid, with an x-range of 12 to 15,
    >>> # and a y-range of 21 to 24
    >>> new_grid = pygmt.grdcut(grid=grid, region=[12, 15, 21, 24])
```
This suggestion should fix the failing tests due to the second region not being in the first.

Comment (willschlitzer) [reply pygmt/src/grdcut.py]: :facepalm: numbers are difficult

**Comment by meghanrjones** [comment]
The example looks nice! 

Do you want to figure out the CI separation mentioned [here](https://github.com/GenericMappingTools/pygmt/issues/1666#issuecomment-993962742) as part of this RFC?

**Comment by willschlitzer** [comment]
> The example looks nice!
> 
> Do you want to figure out the CI separation mentioned [here](https://github.com/GenericMappingTools/pygmt/issues/1666#issuecomment-993962742) as part of this RFC?

@meghanrjones I decided to use `# doctest: +SKIP`; I think the better long term solution is to edit the makefile, but I don't want to go outside the scope of this PR.

**Comment by meghanrjones** [comment]
> > The example looks nice!
> > Do you want to figure out the CI separation mentioned [here](https://github.com/GenericMappingTools/pygmt/issues/1666#issuecomment-993962742) as part of this RFC?
> 
> @meghanrjones I decided to use `# doctest: +SKIP`; I think the better long term solution is to edit the makefile, but I don't want to go outside the scope of this PR.

It's nice that the doctest comments are not included in the rendered html docs: https://pygmt-5z64fvi6h-gmt.vercel.app/api/generated/pygmt.grdcut.html#pygmt.grdcut.

**Review by meghanrjones — APPROVED** [review]
Looks good to me!

**Review by weiji14 — APPROVED** [review]
> > > The example looks nice!
> > > Do you want to figure out the CI separation mentioned [here](https://github.com/GenericMappingTools/pygmt/issues/1666#issuecomment-993962742) as part of this RFC?
> > 
> > @meghanrjones I decided to use `# doctest: +SKIP`; I think the better long term solution is to edit the makefile, but I don't want to go outside the scope of this PR.
> 
> It's nice that the doctest comments are not included in the rendered html docs: https://pygmt-5z64fvi6h-gmt.vercel.app/api/generated/pygmt.grdcut.html#pygmt.grdcut.

Wow, very handy that they hide the `# doctest: +SKIP` comment! So much for overthinking. Let's hope that Sphinx keeps this as a 'feature' rather than a 'bug' :laughing:

**Comment by seisman** [comment]
Is there any way to skip the entire code block using a single directive?

**Comment by willschlitzer** [comment]
> Is there any way to skip the entire code block using a single directive?

The example I found online involves wrapping the example in a function, and then calling that function but then adding `doctest: +SKIP`. I think that would complicate the example code if we tried that.

**Review by seisman — APPROVED** [review]

**Pull request merged**